ThingIFSDK/ThingIFSDK/QueryClause.swift

```diff
@@ -0,0 +1,368 @@
+//
+//  QueryClause.swift
+//  ThingIFSDK
+//
+//  Created on 2017/01/23.
+//  Copyright (c) 2017 Kii. All rights reserved.
+//
+
+import Foundation
+
+/** Base class for query clause classes.
+
+ This is abstract class. Developers does not need to directly use this
+ class.
+ */
+open class QueryClause: NSObject, BaseClause {
```

Comment:
I want to do so but swift does not allow it :(

Swift does not allow to use `Protocol` as type parameter.
`AndClauseInQuery` and `OrClauseInQuery` use `QueryClause` as type parameter by `typealias`. We need concrete class.

In my knowledge, this is better way.

What do you think?